core/src/trezor/strings.py

```diff
@@ -0,0 +1,38 @@
+def format_amount(amount: int, decimals: int) -> str:
+    if amount < 0:
+        amount = -amount
+        sign = "-"
+    else:
+        sign = ""
+    d = pow(10, decimals)
+    s = (
+        ("%s%d.%0*d" % (sign, amount // d, decimals, amount % d))
+        .rstrip("0")
+        .rstrip(".")
+    )
+    return s
+
+
+def format_ordinal(number: int) -> str:
+    return str(number) + {1: "st", 2: "nd", 3: "rd"}.get(
+        4 if 10 <= number % 100 < 20 else number % 10, "th"
+    )
+
+
+def format_plural(string: str, count: int, plural: str):
+    """
+    Adds plural form to a string based on `count`.
+    !! Does not work with irregular words !!
+
+    Example:
+    >>> format_plural("We need %(count)s more %(plural)s", 3, "share")
+    >>> 'We need 3 more shares'
+    >>> format_plural("We need %(count)s more %(plural)s", 1, "share")
+    >>> 'We need 1 more share'
+    """
+    if not all(s in string for s in ("%(count)", "%(plural)")):
+        # string needs to have %(count) and %(plural) inside
+        raise ValueError()
```

Comment:
if there are no arguments, it should just be `raise ValueError` without parens

Reply:
I always forget that. Is the reason the without () I'm not creating the object and therefore saving an allocation?

Reply:
the reason is Python style guide :) but yes, what you said is also true